API/MoodisAPI/apiMoodis/apiMoodis/Controllers/ImageInfoController.cs

```diff
@@ -0,0 +1,85 @@
+﻿using apiMoodis.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using System.Data.Entity;
+
+namespace apiMoodis.Controllers
+{
+    public class ImageInfoController : ApiController
+    {
+        [HttpGet]
+        public IHttpActionResult GetAllImageInfos()
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                var imageInfos = dbContext.ImageInfos.Include(e => e.Emotions).ToList();
+                if (imageInfos.Count() == 0)
+                {
+                    return BadRequest("There are no ImageInfos");
+                }
+                else
+                {
+                    return Ok(imageInfos);
+                }
+            }
+        }
+
+        [HttpGet]
+        public IHttpActionResult GetImageInfoById(string id)
+        {
+            using (DatabaseContext dbContext = new DatabaseContext())
+            {
+                var entity = dbContext.ImageInfos.Single(ImageInfo => ImageInfo.Id == id);
+                return Ok(entity);
+            }
+        }
+
+        [HttpPost]
+        public IHttpActionResult PostImageInfo([FromBody] ImageInfo imageInfo)
+        {
+            try
+            {
+                using (DatabaseContext dbContext = new DatabaseContext())
+                {
+                    dbContext.ImageInfos.Add(imageInfo);
```

Comment:
will it add emotions automatically?

Reply:
Yes it will, but only if the emotions that the ImageInfo contains follow the foreign key constraint. If the foreign key does not match but someone tries to POST an ImageInfo that contains invalid emotions for that User, 500 error (exception) will be thrown. Otherwise, all of the objects of the list will be added to the database as appropriate objects. Same thing applies to the User class.